sensorprocess/lidarsensorprocess.go

```diff
@@ -16,128 +16,71 @@ import (
-	if err != nil {
-		return status
-	}
-
-	// update stored lidar timestamp
-	config.updateMutexProtectedLidarData(tsr)
-
-	return false
+	return nil
 }
 
 // tryAddLidarReadingUntilSuccess adds a reading to the cartofacade and retries on error (offline mode). While add lidar
 // reading fails, keep trying to add the same reading - in offline mode we want to process each reading so if we cannot
 // acquire the lock we should try again.
-func (config *Config) tryAddLidarReadingUntilSuccess(ctx context.Context, reading s.TimedLidarReadingResponse) {
+func (config *Config) tryAddLidarReadingUntilSuccess(ctx context.Context, reading s.TimedLidarReadingResponse) error {
 	for {
 		select {
 		case <-ctx.Done():
-			return
+			return ctx.Err()
 		default:
-			err := config.CartoFacade.AddLidarReading(ctx, config.Timeout, config.Lidar.Name(), reading)
-			if err == nil {
+			if err := config.CartoFacade.AddLidarReading(ctx, config.Timeout, config.Lidar.Name(), reading); err != nil {
+				config.Logger.Debugf("%v \t | LIDAR | Failure \t \t | %v \n", reading.ReadingTime, reading.ReadingTime.Unix())
+				if !errors.Is(err, cartofacade.ErrUnableToAcquireLock) {
+					config.Logger.Warnw("Retrying sensor reading due to error from cartofacade", "error", err)
+				}
+			} else {
 				config.Logger.Debugf("%v \t | LIDAR | Success \t \t | %v \n", reading.ReadingTime, reading.ReadingTime.Unix())
-				return
-			}
-			if !errors.Is(err, cartofacade.ErrUnableToAcquireLock) {
-				config.Logger.Warnw("Retrying sensor reading due to error from cartofacade", "error", err)
+				return nil
 			}
-			config.Logger.Debugf("%v \t | LIDAR | Failure \t \t | %v \n", reading.ReadingTime, reading.ReadingTime.Unix())
 		}
 	}
 }
 
-// tryAddLidarReading adds a reading to the carto facade and does not retry (online).
-//
-//nolint:dupl
+// tryAddLidarReading adds a reading to the carto facade and does not retry (online). Returns remainder of time interval.
 func (config *Config) tryAddLidarReading(ctx context.Context, reading s.TimedLidarReadingResponse) int {
 	startTime := time.Now().UTC()
 
-	err := config.CartoFacade.AddLidarReading(ctx, config.Timeout, config.Lidar.Name(), reading)
-	if err != nil {
+	if err := config.CartoFacade.AddLidarReading(ctx, config.Timeout, config.Lidar.Name(), reading); err != nil {
 		config.Logger.Debugf("%v \t | LIDAR | Failure \t \t | %v \n", reading.ReadingTime, reading.ReadingTime.Unix())
 		if errors.Is(err, cartofacade.ErrUnableToAcquireLock) {
 			config.Logger.Debugw("Skipping lidar reading due to lock contention in cartofacade", "error", err)
 		} else {
 			config.Logger.Warnw("Skipping lidar reading due to error from cartofacade", "error", err)
 		}
+	} else {
+		config.Logger.Debugf("%v \t | LIDAR | Success \t \t | %v \n", reading.ReadingTime, reading.ReadingTime.Unix())
 	}
-	config.Logger.Debugf("%v \t | LIDAR | Success \t \t | %v \n", reading.ReadingTime, reading.ReadingTime.Unix())
```

Comment:
I moved this into the `else` statement because otherwise we'd report a success even if there was a failure.